BranchSDK/src/BranchIO/Request.cpp

```diff
@@ -87,6 +96,23 @@ Request::getAttemptCount() const {
     return _attemptCount;
 }
 
+int
+Request::getMaxRetryCount() const {
+    Mutex::ScopedLock _l(_mutex);
+    return _maxRetryCount;
+}
+
+void
+Request::setMaxRetryCount(int retryCount) {
+    // A retry count < 0 has special meaning.
+    if (retryCount < 0) {
+        _maxRetryCount = DefaultRetryCount;
+    } else {
+        // We don't want to try forever... cap at Max.
+        _maxRetryCount = min(DefaultRetryCount, retryCount);
```

Comment:
Should the max cap be a different value/variable, not `DefaultRetryCount`? This basically limits users to setting retry count in the 0-5 range. If that's intended, we should mention it in the method description.

Reply:
Reverted this max and default retry logic.